pkg/controller/volume/persistentvolume/pv_controller_test.go

```diff
@@ -762,7 +787,11 @@ func TestUpdateFinalizer(t *testing.T) {
 			if tc.volumeAnnotations != nil {
 				ann := tc.volumeAnnotations
 				finalizers := tc.volumeFinalizers
-				modified := updateMigrationAnnotationsAndFinalizers(cmpm, translator, ann, &finalizers, false)
+				pvFinalizerModSpec := &pvFinalizerModifySpec{
+					isInTreeVolume:  tc.inTreeVolume,
+					pvReclaimPolicy: v1.PersistentVolumeReclaimDelete,
```

Comment:
Please add tests with `PersistentVolumeReclaimRetain` and `PersistentVolumeReclaimRecycle` and check the finalizers are removed.